### PR TITLE
[PREVIEW] SIDM-3159 ip sanitise

### PR DESCRIPTION
**Pull request by tbamido**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SIDM-3159

### Change description ###
Sanitise ips on policy evaluation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java [hunk 1]
```diff
@@ -104,4 +109,38 @@ private boolean checkNoActionsBlockingUser(ResponseEntity<EvaluatePoliciesRespon
...
+        final String[] splitArray = StringUtils.split(ipAddress, ",");
+        return Arrays.asList(splitArray).stream()
+            .map(String::trim)
+            .map(s -> {
+                final boolean isIpv4 = StringUtils.countMatches(s, ":") < 2;
+                if (isIpv4) {
+                    // remove port if any
+                    return StringUtils.substringBefore(s, ":");
+                }
+                final Matcher m = Pattern.compile("\\[(.+)\\].*").matcher(s);
```
Comment (nikola-naydenov-hmcts) [inline src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java]: make this pattern static, so you dont have to keep compiling it

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java [hunk 2]
```diff
@@ -104,4 +109,38 @@ private boolean checkNoActionsBlockingUser(ResponseEntity<EvaluatePoliciesRespon
...
+     * Sanitise and return request ips in a list of strings
+     * Examples:
+     * "1.1.1.1" => ["1.1.1.1"]
+     * "51.140.12.192:59286" => ["51.140.12.192"]
+     * "51.140.12.192:59286, 10.97.64.4:59250, 10.97.66.7:57249" => ["51.140.12.192", "10.97.64.4", "10.97.66.7"]
+     * "2001:db8:85a3:8d3:1319:8a2e:370:7348" => ["2001:db8:85a3:8d3:1319:8a2e:370:7348"]
+     * "[2001:db8:85a3:8d3:1319:8a2e:370:7348]:1234" => ["2001:db8:85a3:8d3:1319:8a2e:370:7348"]
+     *
+     * @should returnSanitisedIpAddresses
+     */
```
Comment (nikola-naydenov-hmcts) [inline src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java]: name is a bit ambiguous, maybe sanitiseRequestedIps? get indicates you are getting them from somewhere

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java [hunk 3]
```diff
@@ -104,4 +109,38 @@ private boolean checkNoActionsBlockingUser(ResponseEntity<EvaluatePoliciesRespon
         return true;
     }
 
+    /**
+     * Sanitise and return request ips in a list of strings
+     * Examples:
+     * "1.1.1.1" => ["1.1.1.1"]
+     * "51.140.12.192:59286" => ["51.140.12.192"]
+     * "51.140.12.192:59286, 10.97.64.4:59250, 10.97.66.7:57249" => ["51.140.12.192", "10.97.64.4", "10.97.66.7"]
```
Comment (kremi) [inline src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java]: In that case - does AM check that _none_ of the IPs in the list is in the range of allowed IPs?

Comment (tbamido) [reply src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java]: we can have multiple rules
using AND or OR and NOT we can achieve what we need 👍

**Pull request merged**